app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -45,10 +45,8 @@ def authenticate_user
 
   def handle_second_factor_locked_user(type)
     analytics.track_event(Analytics::MULTI_FACTOR_AUTH_MAX_ATTEMPTS)
-
-    sign_out
-
     render 'two_factor_authentication/shared/max_login_attempts_reached', locals: { type: type }
+    sign_out
```

Comment:
thanks for catching this! going to create a GH issue for this now and will assign myself. Agree we can deal with the `nil` issue, but we should also make sure there is a good test alongside it so this bug doesn't get re-introduced!